platform/security/security-oidc-realm/src/main/java/org/codice/ddf/security/oidc/realm/CustomOidcProfileCreator.java

```diff
@@ -47,85 +35,44 @@
-    } catch (java.text.ParseException | ParseException e) {
-      LOGGER.warn("Unable to extract ID token claim set.", e);
+    final AccessToken accessToken = credentials.getAccessToken();
+    if (accessToken != null && !accessToken.getValue().isEmpty()) {
+      profile.setAccessToken(accessToken);
     }
 
     final RefreshToken refreshToken = credentials.getRefreshToken();
     if (refreshToken != null && !refreshToken.getValue().isEmpty()) {
       profile.setRefreshToken(refreshToken);
-      LOGGER.debug("Refresh Token successful retrieved");
     }
 
-    try {
-      if (metadata.getUserInfoEndpointURI() != null && accessToken != null) {
-
-        final UserInfoRequest userInfoRequest =
-            new UserInfoRequest(metadata.getUserInfoEndpointURI(), (BearerAccessToken) accessToken);
-        final HTTPRequest userInfoHttpRequest = userInfoRequest.toHTTPRequest();
-        userInfoHttpRequest.setConnectTimeout(configuration.getConnectTimeout());
-        userInfoHttpRequest.setReadTimeout(configuration.getReadTimeout());
-        final HTTPResponse httpResponse = userInfoHttpRequest.send();
-        LOGGER.debug(
-            "Token response: status={}, content={}",
-            httpResponse.getStatusCode(),
-            httpResponse.getContent());
-
-        final UserInfoResponse userInfoResponse = UserInfoResponse.parse(httpResponse);
-        if (userInfoResponse instanceof UserInfoErrorResponse) {
-          LOGGER.error(
-              "Bad User Info response, error={}",
-              ((UserInfoErrorResponse) userInfoResponse).getErrorObject());
-        } else {
-          final JWTClaimsSet userInfoClaimsSet;
-          final UserInfoSuccessResponse userInfoSuccessResponse =
-              (UserInfoSuccessResponse) userInfoResponse;
-          if (userInfoSuccessResponse.getUserInfo() != null) {
-            userInfoClaimsSet = userInfoSuccessResponse.getUserInfo().toJWTClaimsSet();
-          } else {
-            userInfoClaimsSet = userInfoSuccessResponse.getUserInfoJWT().getJWTClaimsSet();
-          }
+    final JWT idToken = credentials.getIdToken();
+    profile.setIdTokenString(idToken.getParsedString());
 
-          getProfileDefinition().convertAndAdd(profile, userInfoClaimsSet.getClaims(), null);
-        }
-      }
+    try {
+      JWTClaimsSet claimsSet = idToken.getJWTClaimsSet();
+      assertNotNull("claimsSet", claimsSet);
+      profile.setId(ProfileHelper.sanitizeIdentifier(profile, claimsSet.getSubject()));
 
-      for (final Map.Entry<String, Object> entry :
-          idToken.getJWTClaimsSet().getClaims().entrySet()) {
-        if (!JwtClaims.SUBJECT.equals(entry.getKey())
-            && profile.getAttribute(entry.getKey()) == null) {
-          getProfileDefinition()
-              .convertAndAdd(profile, PROFILE_ATTRIBUTE, entry.getKey(), entry.getValue());
-        }
+      for (final Map.Entry<String, Object> entry : claimsSet.getClaims().entrySet()) {
```

Comment:
📘 This is what is mapping the userinfo/idtoken information to the `OidcProfile`.